django/contrib/admin/options.py

```diff
@@ -896,9 +900,17 @@ def construct_search(field_name):
             orm_lookups = [construct_search(str(search_field))
                            for search_field in search_fields]
             for bit in search_term.split():
-                or_queries = [models.Q(**{orm_lookup: bit})
-                              for orm_lookup in orm_lookups]
+                or_queries = []
+                for orm_lookup in orm_lookups:
+                    if orm_lookup.endswith('__exact'):
+                        # This is an IntegerField.
+                        try:
+                            bit = int(bit)
+                        except ValueError:
+                            break
+                    or_queries.append(models.Q(**{orm_lookup: bit}))
```

Comment:
`IntegerField.get_prep_value` gives an error when I try to search a non-integer field like _"a10"_ from the search field. This was to ensure that.